contracts/ts/genMaciState.ts

```diff
@@ -449,18 +449,6 @@ const genMaciStateFromContract = async (
       maciState.polls[pollId].publishMessage(action.data.message, action.data.encPubKey);
     } else if (action["type"] === "TopupMessage") {
       maciState.polls[pollId].topupMessage(action.data.message);
-    } else if (action["type"] === "MergeMaciStateAqSubRoots") {
-      maciState.stateAq.mergeSubRoots(action.data.numSrQueueOps);
-    } else if (action["type"] === "MergeMaciStateAq") {
-      if (pollId == 0) {
-        maciState.stateAq.merge(stateTreeDepth);
-      }
-    } else if (action["type"] === "MergeMessageAqSubRoots") {
-      maciState.polls[pollId].messageAq.mergeSubRoots(action.data.numSrQueueOps);
-    } else if (action["type"] === "MergeMessageAq") {
-      maciState.polls[pollId].messageAq.merge(treeDepths.messageTreeDepth);
-      const poll = maciState.polls[pollId];
-      assert(poll.messageAq.mainRoots[treeDepths.messageTreeDepth] === action.data.messageRoot);
```

Comment:
[This test case](https://github.com/privacy-scaling-explorations/maci//blob/newstr/contracts/tests/MACI.test.ts#L397-L415) from `maci-contracts` checks it

Reply:
sure, but aren't these two different files? `genMaciState` deals with with fetching logs from an RPC and passing them to create a local `MaciState`. What if something was missing in the logs (very pessimistic view ofc) and we create the wrong state? we could catch it here by checking against the on chain root

Reply:
Both `MergeMessageAq` and `mergeMessageAqSubRoots` are helper commands and not essential part of the core data structurue(i.e. quinary merkle tree). It's meant to manage the interim state of the accumulator queue, which has been introduced to manage gas efficiency onchain. That said, they doesn't affect the tree root.

Reply:
Although, I think we should keep the all the logs in the state machine instance. It should be useful in certain cases like debugging. I will file an issue about this

Reply:
Sorry, I think I might have phrased my comment wrong. 
What I am suggesting is to still remove the part of the code you removed, however keep the check you are doing (just move it down), but do it on the on chain root vs the local maci state state tree (the quinary tree). Hope this is a bit more clear. So this would be assert(MACI state tree on-chain root === `poll.stateTree.root`)

Reply:
Fair enough. Ideally, it should be ensured by the test cases of [OptimisedMT](https://github.com/weijiekoh/optimisedmt/blob/main/ts/__tests__/OptimisedMT.test.ts. There's nothing harmful in having a safeguard though